pkg/fleet/telemetry/span.go

```diff
@@ -7,20 +7,146 @@
 package telemetry
 
 import (
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"context"
+	"fmt"
+	"math/rand/v2"
+	"runtime/debug"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/internaltelemetry"
 )
 
-// Span is an alias for ddtrace.Span until we phase ddtrace out.
-type Span struct{ ddtrace.Span }
+const spanKey = spanContextKey("span_context")
+
+type spanContextKey string
+
+// Span represents a span.
+type Span struct {
+	mu       sync.Mutex
+	span     internaltelemetry.Span
+	finished atomic.Bool
+}
+
+func newSpan(name string, parentID, traceID uint64) *Span {
+	var noParent bool
+	if traceID == 0 {
+		traceID = rand.Uint64()
+		if !headSamplingKeep(name, traceID) {
+			traceID = dropTraceID
+		}
+		noParent = true
```

Comment:
Shouldn't this be set if `parentID == 0` instead?

Reply:
both work, it is nicer with parentID